x-pack/legacy/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.ts

```diff
@@ -82,14 +83,31 @@ export function initSpacesOnPostAuthRequestInterceptor({
 
         space = await spacesClient.get(spaceId);
       } catch (error) {
-        log.error(
-          `Unable to navigate to space "${spaceId}", redirecting to Space Selector. ${error}`
-        );
-        // Space doesn't exist, or user not authorized for space, or some other issue retrieving the active space.
-        const result = response.redirected({
-          headers: { location: getSpaceSelectorUrl(serverBasePath) },
-        });
-        return result;
+        const wrappedError = wrapError(error);
+
+        const statusCode = wrappedError.output.statusCode;
+
+        // If user is not authorized, or the space cannot be found, allow them to select another space
+        // by redirecting to the space selector.
+        const shouldRedirectToSpaceSelector = statusCode === 403 || statusCode === 404;
+
+        if (shouldRedirectToSpaceSelector) {
+          log.error(
```

Comment:
```suggestion
          log.debug(
```

Should we consider downgrading this from `error` to `debug`, since this is expected now?

Reply:
hmm, I still think of this as an error condition -- the user attempted to navigate to a space, and it couldn't happen for one reason or another. The list of possible reasons to redirect to the space selector has been reduced as part of this PR, but I think the spirit of the operation is still much the same.

The expected paths that produce a `403` and `404` error here are well known to us, so I can see your argument for reducing the severity, but these could also be thrown for reasons yet unknown, or unknown to the Elastic Stack components (a proxy or other misconfigured network appliance could be there, for example). 

Having this surfaced as an error could help quickly diagnose these types of problems without turning on debug logging, as this interceptor will be one of the very first code paths to query Elasticsearch during a user's request lifecycle.

Reply:
There's an inconsistency here though. Anytime that we try to navigate to a space where the user is unauthorized or it just doesn't exist, we log at the `error` level. We don't do this anywhere else. 

>  these could also be thrown for reasons yet unknown, or unknown to the Elastic Stack components (a proxy or other misconfigured network appliance could be there, for example).
> 
> Having this surfaced as an error could help quickly diagnose these types of problems without turning on debug logging, as this interceptor will be one of the very first code paths to query Elasticsearch during a user's request lifecycle.

This seems like an edge-case. If we're looking to investigate a scenario like this, we'd likely already be requesting logs that include the debug level.

Reply:
> There's an inconsistency here though. Anytime that we try to navigate to a space where the user is unauthorized or it just doesn't exist, we log at the error level. We don't do this anywhere else.

Should we make this a `debug` condition too then?
https://github.com/elastic/kibana/blob/d3644b3514209d7500d4745f912d4e4b0f6b66e2/x-pack/legacy/plugins/spaces/server/lib/request_interceptors/on_post_auth_interceptor.ts#L130

Reply:
That'd be my preference, but I don't wanna be forcing my opinion on you. I primarily looking at the security plugin itself for the precedent, and didn't realize we did it here as well.